tests/stability/test_adaptive_scaling.py

```diff
@@ -0,0 +1,302 @@
+            time.sleep(adapt.interval * 2.1)  # Ensure enough time for system to adapt
+            assert len(adapt.log) == 0
+            ev_fan_out = Event(name="fan-out", client=client)
+
+            def clog(x: int, ev: Event) -> int:
+                ev.wait()
+                return x
+
+            futures = client.map(clog, range(100), ev=ev_fan_out)
+
+            start = time.monotonic()
+            client.wait_for_workers(n_workers=maximum, timeout=TIMEOUT_THRESHOLD)
+            end = time.monotonic()
+            duration = end - start
+            assert duration < threshold, duration
+            assert len(adapt.log) <= 2
+            assert adapt.log[-1][1] == {"status": "up", "n": maximum}
+            ev_fan_out.set()
+            client.gather(futures)
+            return duration
+
+
+@pytest.mark.skip(reason="coiled-runtime#266")
+@pytest.mark.stability
+def test_adapt_to_changing_workload():
+    """Tests that adaptive scaling reacts within a reasonable amount of time to
+    a varying task load and scales up or down. This also asserts that no recomputation
+    is caused by the scaling.
+    """
+    minimum = 0
+    maximum = 10
+    fan_out_size = 100
+    with Cluster(
+        name=f"test_adaptive_scaling-{uuid.uuid4().hex}",
+        n_workers=maximum,
+        worker_vm_types=["t3.medium"],
+        wait_for_workers=True,
+        # Note: We set allowed-failures to ensure that no tasks are not retried upon ungraceful shutdown behavior
+        # during adaptive scaling but we receive a KilledWorker() instead.
+        environ={"DASK_DISTRIBUTED__SCHEDULER__ALLOWED_FAILURES": "0"},
+    ) as cluster:
+        with Client(cluster) as client:
+
+            @delayed
+            def clog(x: int, ev: Event, **kwargs) -> int:
+                ev.wait()
+                return x
+
+            def workload(
+                fan_out_size,
+                ev_fan_out,
+                ev_barrier,
+                ev_final_fan_out,
+            ):
+                fan_out = [clog(i, ev=ev_fan_out) for i in range(fan_out_size)]
+                barrier = clog(delayed(sum)(fan_out), ev=ev_barrier)
+                final_fan_out = [
+                    clog(i, ev=ev_final_fan_out, barrier=barrier)
+                    for i in range(fan_out_size)
+                ]
+                return final_fan_out
+
+            ev_fan_out = Event(name="fan-out", client=client)
+            ev_barrier = Event(name="barrier", client=client)
+            ev_final_fan_out = Event(name="final-fan-out", client=client)
+
+            fut = client.compute(
+                workload(
+                    fan_out_size=fan_out_size,
+                    ev_fan_out=ev_fan_out,
+                    ev_barrier=ev_barrier,
+                    ev_final_fan_out=ev_final_fan_out,
+                )
+            )
+
+            adapt = cluster.adapt(minimum=minimum, maximum=maximum)
+            wait(90)
```

Comment:
Does this work? I'd think you have to pass some futures in here or something. Did you mean to sleep here?